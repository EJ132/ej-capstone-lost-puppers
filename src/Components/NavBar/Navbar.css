/* MOBILE */

.navbar {
    background-color: black;
    width: 100%;
    color: white;
    display: flex;
    flex-direction: row;
    list-style-type: none;
    padding: 2px;
    justify-content: flex-end;
    position: fixed;
    top: 0;
    left: 0;
    height: 40px;
    z-index: 9;
}

.navbar li {
    padding: 6px 20px 6px 6px;
    font-size: 10px;
    margin-top: 5px;
}

.navbar a {
    text-decoration: none;
    color: white;
}

.navbar a:visited {
    text-decoration: none;
    color: white;
}

.navbar img {
    position: absolute;
    width: 30px;
    height: 30px;
    left: 10px;
    top: 5px;
}

/* DESKTOP */

@media screen and (min-width: 768px){
    .navbar {
        padding: 12px;
        height: 50px;
    }
    
    .navbar li {
        font-size: 1rem;
        margin-top: 0px;
    }
    
    .navbar img {
        width: 40px;
        height: 40px;
        left: 10px;
        top: 5px;
    }
}