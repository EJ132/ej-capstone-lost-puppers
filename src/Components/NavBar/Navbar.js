import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import './Navbar.css';
import paw from '../../Images/paw.png';
import TokenService from '../../services/token-service'

export default class Navbar extends Component {

handleLogoutClick = () => {
    TokenService.clearAuthToken()
    TokenService.clearUserName()
    }

    renderLogoutLink() {
    return (
        <Link
            onClick={this.handleLogoutClick}
            to='/'>
            Logout
        </Link>
    )
    }

    renderLoginLink() {
    return (
        <Link
            to='/login'>
            Log in
        </Link>
    )
    }

    render () {
        
        return (
            <ul className="navbar">
                <li><Link to="/">Home</Link></li>
                <li><Link to="/aboutus">About Us</Link></li>
                <li><Link to="/find">Find a Pupper</Link></li>
                {TokenService.hasAuthToken() ? <li><Link to='/profile'>Profile</Link></li> : null}
                <li>{TokenService.hasAuthToken()
                    ? this.renderLogoutLink()
                    : this.renderLoginLink()}</li>
                <img alt="paw" src={paw}></img>
            </ul>
        )
    }
}