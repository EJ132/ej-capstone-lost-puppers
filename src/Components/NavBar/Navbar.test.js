import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NavBar from './Navbar'

describe(`Navbar component`, () => {

  it('renders the NavBar', () => {
    const wrapper = shallow(<NavBar />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
