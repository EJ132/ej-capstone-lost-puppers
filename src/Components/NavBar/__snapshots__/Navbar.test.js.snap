// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders the NavBar 1`] = `
<ul
  className="navbar"
>
  <li>
    <Link
      to="/"
    >
      Home
    </Link>
  </li>
  <li>
    <Link
      to="/aboutus"
    >
      About Us
    </Link>
  </li>
  <li>
    <Link
      to="/find"
    >
      Find a Pupper
    </Link>
  </li>
  <li>
    <Link
      to="/login"
    >
      Log in
    </Link>
  </li>
  <img
    alt="paw"
    src="paw.png"
  />
</ul>
`;
